2025-07-15 00:32:28,437 : ERROR : Exception on /explore [GET] [in C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py : 875]
Traceback (most recent call last):
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\app\routes.py", line 167, in explore
    return render_template('index.html', posts=posts.items, title = 'explore', next_url = next_url, prev_url= prev_url)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 1295, in render
    self.environment.handle_exception()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Kakashi\Desktop\projects\newblog\app\templates\index.html", line 2, in top-level template code
    {% import 'bootstrap_wtf.html' as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\app\templates\base.html", line 58, in top-level template code
    {% block body %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\app\templates\index.html", line 8, in block 'body'
    {{ wtf.quick_form(form, action=url_for('index')) }}
    ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\runtime.py", line 784, in _invoke
    rv = self._func(*arguments)
  File "C:\Users\Kakashi\Desktop\projects\newblog\app\templates\bootstrap_wtf.html", line 60, in template
    {{ form.hidden_tag() }}
    ^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 490, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
2025-07-15 00:33:46,148 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 30]
2025-07-15 00:46:24,641 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 30]
2025-07-15 01:07:46,016 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 30]
2025-07-15 01:09:30,507 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 30]
2025-07-15 22:44:57,113 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 32]
2025-07-15 22:54:27,425 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 32]
2025-07-17 07:34:13,621 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 32]
